Lib/test/test_asyncio/test_tasks.py

```diff
@@ -2689,13 +2696,15 @@ class PyTask_CFuture_Tests(BaseTaskTests, test_utils.TestCase):
 
     Task = tasks._PyTask
     Future = getattr(futures, '_CFuture', None)
+    all_tasks = getattr(tasks, '_c_all_tasks', None)
 
 
 class PyTask_PyFuture_Tests(BaseTaskTests, SetMethodsTest,
```

Comment:
there are sub-classes for all combinations of {C,Py}-{Task,Future} -- should `all_tasks` be another dimension in this matrix? (and double the number of test cases?)
or are you explicitly coupling `all_tasks` to `Tasks`?

Reply:
@kumaraditya303 see this question

Reply:
Tests which use task implemented in C use the c implementation, rest use the pure python one.